### PR TITLE
Glob calls are now cached

### PR DESCRIPTION
**Pull request by yipstanley**
## Summary:

Glob calls now use the asynchronous function as opposed to the synchronous `glob.sync` function. The asynchronous function allows us to save the cache object from the previous call and pass it into the next call, saving us `fs` calls, which seems to speed things up by a factor of ~1.2x in small tests (see `lib/__test__/utils.test.js`). Also, cleaned up some code comments, tests, and variable names.

Issue: WEB-2605

## Test plan:

The unit tests in `lib/__test__/utils.test.js` compare the speed of the synchronous and asynchronous versions.

### CODE REVIEW

File: js/execCmd.js [hunk 1]
```diff
@@ -0,0 +1 @@
+var _interopRequireDefault=require("@babel/runtime/helpers/interopRequireDefault");var _regenerator=_interopRequireDefault(require("@babel/runtime/regenerator"));var exec=require('@actions/exec');var execCmd=function execCmd(cmd,args){var output,error,options;return _regenerator.default.async(function execCmd$(_context){while(1){switch(_context.prev=_context.next){case 0:output='';error='';options={};options.listeners={stdout:function stdout(data){output+=data.toString();},stderr:function stderr(data){error+=data.toString();}};_context.next=6;return _regenerator.default.awrap(exec.exec(cmd,args,options));case 6:if(!error){_context.next=8;break;}throw error;case 8:return _context.abrupt("return",output);case 9:case"end":return _context.stop();}}},null,null,null,Promise);};module.exports={execCmd:execCmd};
```
Comment (jaredly) [inline js/execCmd.js:1]: were these included by accident?

File: lib/__test__/utils.test.js [hunk 2]
```diff
@@ -350,3 +338,43 @@ describe('test get file diffs', () => {
         expect(result['otherFile.js']).toEqual(mockOtherFileDiff);
     });
 });
+
+describe('test caching glob calls', () => {
```
Comment (jaredly) [inline lib/__test__/utils.test.js:342]: thanks for adding tests! It looks like they're not yet run by a github action?

File: js/execCmd.js (hunk 1 above)
Comment (yipstanley) [reply js/execCmd.js]: Yep! These were a remnant of the Git history magic that we did. I've removed them.

File: lib/__test__/utils.test.js (hunk 2 above)
Comment (yipstanley) [reply lib/__test__/utils.test.js:342]: I added the github actions in the other PR #2

Comment (jaredly) [reply lib/__test__/utils.test.js:342]: oh interesting. Can you change the base of this PR to be `WEB-2645`? Then we should get the checks here

Comment (yipstanley) [reply lib/__test__/utils.test.js:335]: Okay, I've updated the base of the PR and got the checks to run and mostly pass, but the two tests that are failing right now are failing on really strange exit codes and I'm not quite sure why. The autofix test is throwing an error.

When I run `node ./node_modules/actions-utils/list-changed-files.js | grep '.*\.js$' | xargs npm run -s format-files` in my terminal, it says `error: malformed object name HEAD~20`. That line comes from the first command run in the pipeline.

**Pull request merged**